docs/evaluation/regrid.rst

```diff
@@ -18,5 +14,137 @@ from the
 :py:class:`~sherpa.models.model.RegriddableModel1D`
 or
 :py:class:`~sherpa.models.model.RegriddableModel2D` classes.
-At present the only documentation is provided in the
-:doc:`low-level API module <../model_classes/regrid>`.
+The idea is to create a copy of the model which evaluates the
+model on the new grid, and then resamples it onto the output grid.
+This is handled by the `regrid` method from these two
+classes, which is given the grid to evaluate the model on, and it
+returns a copy of the original model.
+
+The examples assume the following:
+
+    >>> import numpy as np
+
+One dimensional models
+======================
+
+Non-integrated models
+---------------------
+
+For this example we shall evaluate the model on the
+grid x=4450.25, 4450.75, ... 4549.75.
+
+::
+
+   >>> xgrid = np.arange(4450.25, 4550, 0.5)
+
+The "regridded" model is created by passing the grid
+to the :py:meth:`~sherpa.models.model.RegriddableModel1D.regrid`
+method::
+
+   >>> from sherpa.models.basic import Gauss1D
+   >>> mdl = Gauss1D('orig')
+   >>> rmdl = mdl.regrid(xgrid)
+
+The "new" model is really just a wrapper around the original model,
+which means that it shares the parameter values and settings. This
+is shown below where the ``pos`` and ``fwhm`` parameters of the
+``rmdl`` are labeled by the label given to ``mdl`` (namely
+"orig")::
+
+   >>> mdl.pos = 4500
+   >>> mdl.fwhm = 20
+   >>> print(mdl)
+   orig
+      Param        Type          Value          Min          Max      Units
+      -----        ----          -----          ---          ---      -----
+      orig.fwhm    thawed           20  1.17549e-38  3.40282e+38
+      orig.pos     thawed         4500 -3.40282e+38  3.40282e+38
+      orig.ampl    thawed            1 -3.40282e+38  3.40282e+38
+   >>> print(rmdl)
+   regrid1d(orig)
+      Param        Type          Value          Min          Max      Units
+      -----        ----          -----          ---          ---      -----
+      orig.fwhm    thawed           20  1.17549e-38  3.40282e+38
+      orig.pos     thawed         4500 -3.40282e+38  3.40282e+38
+      orig.ampl    thawed            1 -3.40282e+38  3.40282e+38
+
+The regridded model can be evaluated as any other model, but behind the
+scenes the data is calculated on the original grid (in this case ``xgrid``)
+and then converted to match the evaluation grid.
+
+::
+
+   >>> x = np.arange(4460, 4540, 2)
+   >>> y1 = mdl(x)
+   >>> y2 = rmdl(x)
+   >>> np.testing.assert_allclose(y1, y2)
+
+Integrated models
+-----------------
+
+The approach is similar, except that the
+:py:meth:`~sherpa.models.model.RegriddableModel1D.regrid` method is
+given two arrays, for the low and high edges of each bin.
+
+::
+
+   >>> imdl = mdl.regrid(xgrid[:-1], xgrid[1:])
+   >>> y1 = mdl(x[:-1], x[1:])
+   >>> y2 = imdl(x[:-1], x[1:])
+   >>> np.testing.assert_allclose(y1, y2, rtol=1e-2)
+
+The fact that the models are integrated across the bins mean that the
```

Comment:
I think it should be 'means' rather than 'mean'

Reply:
Done